source/src/main/java/com/adataptivescale/rosetta/source/core/SourceGeneratorFactory.java

```diff
@@ -1,37 +1,60 @@
 import com.adataptivescale.rosetta.source.core.interfaces.ColumnExtractor;
 import com.adataptivescale.rosetta.source.core.interfaces.Generator;
+import lombok.extern.slf4j.Slf4j;
 
+import java.lang.reflect.InvocationTargetException;
 import java.util.Collection;
+import java.util.Optional;
 
+@Slf4j
 public class SourceGeneratorFactory {
 
     public static Generator<Database, Connection> sourceGenerator(Connection connection) {
         return sourceGenerator(connection, new DriverManagerDriverProvider());
     }
 
-    public static Generator<Database, Connection> sourceGenerator(Connection connection, JDBCDriverProvider driverProvider) {
-        TablesExtractor tablesExtractor = new TablesExtractor();
+    private static ColumnsExtractor loadColumnExtractor(Connection connection) {
+        Optional<Class<?>> columnExtractorModule = ModuleLoader.loadModuleByAnnotationClassValues(
+                ColumnsExtractor.class.getPackageName(), RosettaModuleTypes.COLUMN_EXTRACTOR, connection.getDbType());
+        if(columnExtractorModule.isEmpty()) {
+            log.warn("Columns extractor not supported for database type: {} falling back to default. ", connection.getDbType());
+            return new ColumnsExtractor(connection);
+        }
+        try {
+            return (ColumnsExtractor) columnExtractorModule.get().getDeclaredConstructor(
+                    Connection.class).newInstance(connection);
+        } catch (InstantiationException | IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static TablesExtractor loadTableExtractor(Connection connection) {
         ColumnExtractor<java.sql.Connection, Collection<Table>> columnsExtractor;
-        if ("bigquery".equals(connection.getDbType())) {
-            columnsExtractor = new BigQueryColumnsExtractor(connection);
-        } else if ("mysql".equals(connection.getDbType())) {
-            columnsExtractor = new MySQLColumnsExtractor(connection);
-        } else if ("snowflake".equals(connection.getDbType())) {
-            columnsExtractor = new SnowflakeColumnsExtractor(connection);
-        } else if ("kinetica".equals(connection.getDbType())) {
-            columnsExtractor = new KineticaColumnsExtractor(connection);
-        } else if ("spanner".equals(connection.getDbType())) {
-            columnsExtractor = new SpannerColumnsExtractor(connection);
-        } else {
-            columnsExtractor = new ColumnsExtractor(connection);
+        Optional<Class<?>> tableExtractorModule = ModuleLoader.loadModuleByAnnotationClassValues(
+                TablesExtractor.class.getPackageName(), RosettaModuleTypes.TABLE_EXTRACTOR, connection.getDbType());
+        if(tableExtractorModule.isEmpty()) {
+            log.warn("Table extractor not supported for database type: {} falling back to default.", connection.getDbType());
```

Comment:
I am not sure if the best option here is to log a WARN, or find some other approach to inform the user.

At the moment I don't have any better recommendation